File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cpp
#include <string>
#include <vector>

using namespace std;

int getDir(string s) {
    if(s == "up") return 0;
    if(s == "right") return 1;
    if(s == "down") return 2;
    if(s == "left") return 3;
    return 0;
}

vector<int> solution(vector<string> keyinput, vector<int> board) {
    int dy[] = {1, 0, -1, 0};
    int dx[] = {0, 1, 0, -1};
    int x = 0, y = 0;
    int my = (int)board[1] / 2;
    int mx = (int)board[0] / 2;
    for(int i = 0; i < keyinput.size(); i++) {
        int dir = getDir(keyinput[i]);
        int ny = y + dy[dir];
        int nx = x + dx[dir];
        
        if(ny >= -my && ny <= my && nx >= -mx && nx <= mx) {
            y = ny;
            x = nx;
        }
    }
    return {x, y};
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.swift
import Foundation

func solution(_ keyinput:[String], _ board:[Int]) -> [Int] {
    let dictDirX: [String: Int] = [
        "left": -1,
        "right": 1,
        "up": 0,
        "down": 0
    ]
    let dictDirY: [String: Int] = [
        "left": 0,
        "right": 0,
        "up": 1,
        "down": -1
    ]
    var y = 0, x = 0
    var ny: Int
    var nx: Int
    var my = board[1] / 2, mx = board[0] / 2
    for key in keyinput {
        ny = y + dictDirY[key]!
        nx = x + dictDirX[key]!
        
        if ny >= -my && ny <= my && nx >= -mx && nx <= mx {
            y = ny
            x = nx
        }
    }
    return [x, y]
}